cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -16,49 +16,58 @@ transferred to the next job in sequence until the last job in the sequence is ex
 failure, the execution is stopped at the failed job and no subsequent jobs in the sequence
 are executed.
 
-To process MapReduce or Spark programs in sequence, specify
-``addWorkflow()`` in your application::
+To execute MapReduce or Spark programs in a Workflow, you will need to add them in your
+application along with the Workflow. You can optionally add a schedule (such as a
+`crontab schedule <../../reference-manual/javadocs/co/cask/cdap/api/app/AbstractApplication.html#scheduleWorkflow(java.lang.String,%20java.lang.String,%20java.lang.String)>`__)
+to the Workflow::
 
   public void configure() {
     ...
-    addWorkflow(new PurchaseHistoryWorkflow());
+    addMapReduce(new MyMapReduce());
+    addMapReduce(new AnotherMapReduce());
+    addSpark(new MySpark());
+    addWorkflow(new MyWorkflow());
+    scheduleWorkflow("FiveHourSchedule", "0 */5 * * *", "MyWorkflow");
+    ...
+  }
 
-You'll then implement the ``Workflow`` interface, which requires the
-``configure()`` method. From within ``configure``, call the
-``addSchedule()`` method to run a WorkFlow job periodically::
+You'll then extend the ``AbstractWorkflow`` class and implement the
+``configure()`` method. Inside ``configure``, you can add multiple MapReduce, Spark, or Custom Action
+programs to the Workflow. The programs will be executed in the order they are specified in the ``configure``
+method::
 
-  public static class PurchaseHistoryWorkflow implements Workflow {
+  public static class MyWorkflow extends AbstractWorkflow {
 
     @Override
-    public WorkflowSpecification configure() {
-      return WorkflowSpecification.Builder.with()
-        .setName("PurchaseHistoryWorkflow")
-        .setDescription("PurchaseHistoryWorkflow description")
-        .startWith(new PurchaseHistoryBuilder())
-        .last(new PurchaseTrendBuilder())
-        .addSchedule(new DefaultSchedule("FiveMinuteSchedule", "Run every 5 minutes",
-                     "0/5 * * * *", Schedule.Action.START))
-        .build();
+    public void configure() {
+        setName("MyWorkflow");
+        setDescription("MyWorkflow description");
+        addMapReduce("MyMapReduce");
+        addSpark("MySpark");
+        addMapReduce("AnotherMapReduce");
+        addAction(new MyAction());
     }
   }
 
-If there is only one MapReduce or Spark program to be run as a part of a WorkFlow,
-use the ``onlyWith()`` method after ``setDescription()`` when building
-the Workflow::
+In this example, the ``MyWorkflow`` will be executed every 5 hours. During each execution of the Workflow,
+the ``MyMapReduce``, ``MySpark``, ``AnotherMapReduce``, and ``MyAction`` programs will be executed in order.
+
```

Comment:
the ``MyMapReduce``, ``MySpark``, and ``AnotherMapReduce`` programs and the ``MyAction`` custom action will be executed in order.